### ISSUE TITLE
Epoch 00046: early stopping on traning models

### ISSUE THREAD

**Issue by 12343954** [zzh8829/yolov3-tf2#187]
```
Epoch 00046: saving model to checkpoints/yolov3_train_46.tf
11/11 [==============================] - 13s 1s/step - loss: 18.6522 - yolo_output_0_loss: 0.0844 - yolo_output_1_loss: 1.1834 - yolo_output_2_loss: 6.4805 - val_loss: 50.8860 - val_yolo_output_0_loss: 0.0998 - val_yolo_output_1_loss: 16.9227 - val_yolo_output_2_loss: 22.9586
Epoch 00046: early stopping
```

what's means of **“early stopping”**？what can i do?

what does these means？why show out these losses?
```
loss
yolo_output_0_loss
yolo_output_1_loss
yolo_output_2_loss
val_loss
val_yolo_output_0_loss
val_yolo_output_1_loss
val_yolo_output_2_loss
```

i trained 100 annotated images with 2 classes.

**Comment by PieroCV** [comment]
Early Stopping is a callback from tensorflow 2.x (Keras API). It means that val_loss will improve until some epoch. Then, Early Stopping will stop training in order to avoid overfitting.

Log messages like "StartAbort Out of range" are tensorflow log messages, just internal calculations.

Now, for yolo losses, each output of Yolov3 has a function. Now is a custom loss, based on sparse categorical crossentropy. To get a good model, val_loss must be under 20.

**Comment by 12343954** [comment]
@PieroCV thank you man!

![image](https://user-images.githubusercontent.com/1804003/75933136-253b7800-5eb4-11ea-8da7-8761968c3aba.png)

but i don't think `[[{{node IteratorGetNext}}]]`、`[[IteratorGetNext/_4]]` these are the log messages.
in the end of each epoch, the last step cann't execute completely. there is something wrong in here, i think. this may be cause the `early stopping`. TF2.1.0 is still show this issus, someone says `using the tf-nightly` can solve it, but i haven't installed successfully yet .

**Comment by 12343954** [comment]
The only I can do is just keep increasing the training image dataset.

![image](https://user-images.githubusercontent.com/1804003/75945564-92abd080-5ed5-11ea-9d91-ce9b7dab0170.png)

**Comment by PieroCV** [comment]
@12343954 
The EarlyStopping verifies if val_loss is improving or not. When it starts to getting worse, it means that the model is overfitting. The EarlyStopping is not an issue, but if you want to, you could modify the code in order to change the patiente of the callback or just remove it. I don't really recommend it, but:
```python
model.compile(optimizer=optimizer, loss=loss,
                      run_eagerly=(FLAGS.mode == 'eager_fit'))

        callbacks = [
            ReduceLROnPlateau(verbose=1),
            EarlyStopping(patience=4, verbose=1), # <== You could modify the patience (4 or even more) or just comment this line
            ModelCheckpoint('checkpoints/yolov3_train_{epoch}.tf',
                            verbose=1, save_weights_only=True),
            TensorBoard(log_dir='logs')
        ]

        history = model.fit(train_dataset,
                            epochs=FLAGS.epochs,
                            callbacks=callbacks,
                            validation_data=val_dataset)
```

**Comment by 12343954** [comment]
@PieroCV Thank you man！
I hate `early stopping` cause i can do nothing for it.
so i had given up `Tensorflow`, i already compiled `Darknet` with the high accuracy and low level CUP&GPU occupancy and very small training set(400 annotated images) and 30+ fps performance. it's wonderful and that's what i want!

![image](https://user-images.githubusercontent.com/1804003/76583496-48999f00-6514-11ea-8fa3-181d3ea559af.png)

![image](https://user-images.githubusercontent.com/1804003/76583547-7252c600-6514-11ea-9bd5-b48736b739d2.png)
